phoenix-core/src/main/java/org/apache/phoenix/hbase/index/covered/data/LocalTable.java

```diff
@@ -49,15 +61,53 @@
 public class LocalTable implements LocalHBaseState {
 
   private RegionCoprocessorEnvironment env;
+    private Map<ImmutableBytesPtr, Result> results = new HashMap<>();;
 
   public LocalTable(RegionCoprocessorEnvironment env) {
     this.env = env;
   }
 
+  public void scanCurrentRowStates(Set<ImmutableBytesPtr> rows, Collection<? extends ColumnReference> columns, long ts) throws IOException {
+      Scan s = IndexManagementUtil.newLocalStateScan(Collections.singletonList(columns));
+      FilterList filterList = new FilterList(FilterList.Operator.MUST_PASS_ONE);
+      for (ImmutableBytesPtr row : rows) {
```

Comment:
just fyi, In the past we've found perf issues with FilterLists with enormous numbers of filters (> 5000)